tests/generation/test_single_turn_generation.py

```diff
@@ -18,25 +20,67 @@ def _assert_agent_syntax(agent_file: Path):
     ast.parse(agent_file.read_text(encoding="utf-8"))
 
 
+def _assert_execution_time_within_limit(agent_trace: AgentTrace, expected_execution_time: int) -> None:
+    """Assert that the agent execution time is within the expected threshold.
+
+    Args:
+        agent_trace: The agent trace object
+        expected_execution_time: Maximum allowed execution time in seconds
+    """
+    agent_execution_time_in_seconds = agent_trace.duration.total_seconds()
+    assert agent_execution_time_in_seconds < expected_execution_time, (
+        f"Agent execution time {agent_execution_time_in_seconds:.2f}s exceeded expected "
+        f"threshold of {expected_execution_time}s"
+    )
+
+
+def _assert_num_turns_within_limit(agent_trace: AgentTrace, expected_num_turns: int) -> None:
```

Comment:
This is kind of already covered by https://github.com/mozilla-ai/agent-factory/blob/6f98d2006acd7448645630c409fe4687f4f7ed09/tests/generated_artifacts/test_generated_traces.py#L21-L25

There different thresholds are tried and you can see improvements by the number of test that pass vs xfail